terraform/ec2/linux/main.tf

```diff
@@ -26,6 +26,45 @@ locals {
   private_key_content = var.ssh_key_name != "" ? var.ssh_key_value : tls_private_key.ssh_key[0].private_key_pem
 }
 
+#####################################################################
+# Create EFS
+#####################################################################
+resource "aws_efs_file_system" "efs" {
+  creation_token = "predictable-${random_id.testing_id.hex}"
+  tags           = {
+    Name = "predictable-${random_id.testing_id.hex}"
+  }
+}
+
+resource "aws_efs_mount_target" "mount" {
+  file_system_id = aws_efs_file_system.efs.id
+  subnet_id = aws_instance.cwagent.subnet_id
+  security_groups = [aws_security_group.ec2_security_group.id]
+}
+
+resource "null_resource" "configure_efs" {
+  depends_on = [
+    aws_efs_mount_target.mount,
+    aws_instance.cwagent
+  ]
+
+  connection {
+    type = "ssh"
+    user = "ec2-user"
+    private_key = local.private_key_content
+    host = aws_instance.cwagent.public_ip
+  }
+
+  provisioner "remote-exec" {
+    # https://docs.aws.amazon.com/efs/latest/ug/mounting-fs-mount-helper-ec2-linux.html
+    inline = [
+      "sudo yum install -y amazon-efs-utils",
+      "sudo mkdir ${var.efs_mount_point}",
+      "sudo mount -t efs -o tls ${aws_efs_file_system.efs.dns_name} ${var.efs_mount_point}/",
+    ]
+  }
+}
```

Comment:
This will not work for a distro of linux that does not have yum, e.g.: Ubuntu.

Reply:
Good catch. I'm trying to think of a clean and agnostic way to install `amazon-efs-utils`

Reply:
I added a script to use yum or apt-get to install amazon-efs-utils, still considering other edge cases for linux

Reply:
Added support for `zypper`. Among the three, all linux distros should now be supported